lib/Bio/EnsEMBL/DataCheck/Checks/SegmentationFileHasBigBed.pm

```diff
@@ -0,0 +1,98 @@
+=head1 LICENSE
+
+Copyright [2018] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::SegmentationFileHasBigBed;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+use Bio::EnsEMBL::DataCheck::Utils qw/sql_count/;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME        => 'SegmentationFileHasBigBed',
+  DESCRIPTION => 'Check that every segmentation_file entry which has been used in the current Regulatory Build is linked to a BIGBED entry in the data_file table and exists on disk',
+  GROUPS      => ['funcgen_integrity', 'funcgen_Post_regulatory_build'],
+  DB_TYPES    => ['funcgen'],
+  TABLES      => ['regulatory_build','regulatory_feature'],
+};
+
+sub skip_tests {
+  my ($self) = @_;
+
+  my $sql = q/
+    SELECT COUNT(name) FROM regulatory_build 
+    WHERE is_current=1
+  /;
+
+  if (! sql_count($self->dba, $sql) ) {
+    return (1, 'The database has no regulatory build');
+  }
+}
+
+sub tests {
+  my ($self) = @_;
+  my $data_file_path = '/nfs/panda/ensembl/production/ensemblftp/data_files2/';
+  my $table_name = 'segmentation_file';
+  my $file_type = 'BIGBED';
+  my $species = $self->species;
+  my $core_dba = $self->get_dba($self->species, 'core');
```

Comment:
Thanks a lot, I didn't know about this method. I used get_dba because I copied this from these tests: https://github.com/Ensembl/ensembl-datacheck/blob/master/lib/Bio/EnsEMBL/DataCheck/Checks/AssemblyConsistent.pm#L41 and https://github.com/Ensembl/ensembl-datacheck/blob/master/lib/Bio/EnsEMBL/DataCheck/Checks/CoordSystemConsistent.pm#L40. I guess these tests could be updated for consistency. 
This change is in my new commit.

Reply:
Thanks, good idea to change those other datachecks (I've only added get_dna_dba fairly recently...)